lib/nerve/utils.rb

```diff
@@ -4,5 +4,14 @@ def safe_run(command)
       res = `#{command}`.chomp
       raise "command '#{command}' failed to run:\n#{res}" unless $?.success?
     end
+
+    def responsive_sleep(seconds)
+      nap_time = seconds
+      while nap_time > 0
+        yield if block_given?
+        sleep [nap_time, 1].min
```

Comment:
Ok if I make it an argument to the function that defaults to some number?

Also is that break in a block thing at all idiomatic? I felt really dirty writing it... but it leads to a fairly concise calling convention.